agent/agent.go

```diff
@@ -78,6 +78,23 @@ func (a *Agent) Start(ctx context.Context) error {
 	return err
 }
 
+// RemoveAssignments clean up assignments from Swarm
+func (a *Agent) RemoveAssignments(ctx context.Context) error {
+	select {
+	case <-a.started:
+	default:
+		return errAgentNotStarted
+	}
+
+	if err := a.worker.Assign(ctx, nil); err != nil {
+		return err
+	}
+	if a.worker.TaskSize() == 0 {
```

Comment:
This is fairly racy if there is any delay in removing assignments. I would just recommend calling `worker.Assign` and ensure that works.